Level roles

EarlJr53 opened the pull request:
Sam and I wrote the level roles stuff. Here it be.

EarlJr53 committed:
Added level roles in levels.ts

EarlJr53 committed:
updated levels.ts

EarlJr53 committed:
Merge branch 'master' into Level-Roles

samollari commented:
Closes #27

samollari commented on triggers/triggers/levels.ts:
```diff
@@ -19,9 +20,52 @@ module.exports = <TriggeredCommand> {
+                const member = message.member;
+
+                
+                let newRole = member.guild.roles.cache.filter(role =>
+                    role.name.startsWith('Level') &&
+                    getLevelNumberFromRole(role) <= newLevel)
+                    .sort((a:Role, b:Role) => getLevelNumberFromRole(a) - getLevelNumberFromRole(b))
+                    .last();
+                
+                member.roles.cache.forEach(role =>
+                    (newRole.name.startsWith('Level') && role.id != newRole.id) ?
+                        member.roles.remove(role)
+                      : null
+                );
+                
+                member.roles.add(newRole);
```
Try pulling this out into `util/levels.ts` and import it from there? It should also run that on every start up to update roles if it fell behind

EarlJr53 committed:
Moved setLevelRoles to util/levels.ts

Moved setLevelRoles and dependent function to util/levels.ts. Also implemented it back into triggers/levels.ts. We still have to implement it on program startup.

EarlJr53 commented:
As I mentioned in the latest commit, we still need to implement setLevelRoles on startup before we can close #27 / #34 . I wasn't exactly sure how we wanted to do that, so I committed what I had and we can figure that out soon.

EarlJr53 committed:
Merge branch 'master' into Level-Roles

EarlJr53 commented:
@i-am-gizm0 can we merge this into master for now before we add the on-startup check? And then do that after?

EarlJr53 committed:
Merge branch 'master' into Level-Roles
